app/Services/GraphQL.php

```diff
@@ -15,14 +16,50 @@ public function __construct()
     }
 
     /**
-     * Run a GraphQL query using the client.
+     * Run a GraphQL query using the client
+     * and parse the data result into a convenient format.
      *
      * @param  $query     String
      * @param  $variables Array
-     * @return \Softonic\GraphQL\Response
+     * @param  $queryKey  String
+     * @return array|null
      */
-    public function query($query, $variables)
+    public function query($query, $variables, $queryKey)
     {
-        return $this->client->query($query, $variables);
+        // Use try/catch to avoid any GraphQL related errors breaking the application.
+        try {
+            $response = $this->client->query($query, $variables);
+        } catch (\Exception $exception) {
+            Log::error(
+                'GraphQL request failed. Variables: '.json_encode($variables).' Exception: '.$exception->getMessage()
+            );
+
+            return null;
+        }
+
+        return $response ? array_get($response->getData(), $queryKey) : null;
+    }
+
+    /**
+     * Query for a CampaignWebsite by campaignId field.
+     *
+     * @param  $campaignId String
+     * @return array|null
+     */
+    public function getCampaignWebsiteByCampaignId($campaignId)
+    {
+        $query = '
```

Comment:
@DFurnes I know you'd mentioned a while back potentially making use of dedicated .gql files to contain these queries -- do you think that's worth implementing now? I felt like this format wasn't quite as obnoxious as the HEREDOC

Reply:
Yeah this looks pretty clean! I think the only thing `.gql` files would really get us is syntax highlighting & editor autocomplete, which probably isn't as necessary here as in other applications!